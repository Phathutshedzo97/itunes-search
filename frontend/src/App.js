

import React, { useState } from 'react';
import axios from 'axios';
import 'bootstrap/dist/css/bootstrap.min.css';

function App() {
  const [searchTerm, setSearchTerm] = useState('');
  const [mediaType, setMediaType] = useState('all');
  const [searchResults, setSearchResults] = useState([]);
  const [error, setError] = useState('');
  const [favorites, setFavorites] = useState([]);

  const handleSearch = async () => {
    try {
      const response = await axios.get(`http://localhost:5009/api/search/${searchTerm}/${mediaType}`);
      setSearchResults(response.data);
      setError('');
    } catch (error) {
      setError('Error fetching data from the server');
    }
  };

  const handleAddToFavorites = (item) => {
    setFavorites((prevFavorites) => [...prevFavorites, item]);
  };

  return (
    <div className="container mt-4">
      <h1 className="mb-4">iTunes Search App</h1>
      <div className="row mb-3">
        <div className="col-md-6 col-lg-4">
          <input
            type="text"
            className="form-control"
            placeholder="Enter search term"
            value={searchTerm}
            onChange={(e) => setSearchTerm(e.target.value)}
          />
        </div>
        <div className="col-md-4 col-lg-3">
          <select className="form-control" value={mediaType} onChange={(e) => setMediaType(e.target.value)}>
            <option value="all">All</option>
            <option value="movie">Movie</option>
            <option value="music">Music</option>
            {/* Add other media types here */}
          </select>
        </div>
        <div className="col-md-2 col-lg-2">
          <button className="btn btn-primary w-100" onClick={handleSearch}>
            Search
          </button>
        </div>
      </div>
      {error && <div className="alert alert-danger">{error}</div>}
      <div>
        {searchResults.map((item) => (
          <div key={item.trackId} className="card mb-3">
            <div className="card-body">
              <h3 className="card-title">{item.trackName}</h3>
              {/* Display other relevant information about the item */}
              <button className="btn btn-sm btn-primary" onClick={() => handleAddToFavorites(item)}>
                Add to Favorites
              </button>
            </div>
          </div>
        ))}
      </div>
      <div className="mt-4">
        <h2>Favorites</h2>
        {favorites.map((item) => (
          <div key={item.trackId} className="card mb-3">
            <div className="card-body">
              <h3 className="card-title">{item.trackName}</h3>
              {/* Display other relevant information about the favorite item */}
            </div>
          </div>
        ))}
      </div>
    </div>
  );
}

export default App;
