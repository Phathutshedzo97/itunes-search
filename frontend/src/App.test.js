import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import axios from 'axios';
import App from './App';

// Mock the axios.get function to return a sample response
jest.mock('axios', () => ({
  get: jest.fn(() => Promise.resolve({ data: [] })),
}));

test('renders iTunes Search App title', () => {
  render(<App />);
  const titleElement = screen.getByText(/iTunes Search App/i);
  expect(titleElement).toBeInTheDocument();

  // Create a snapshot of the rendered component
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

test('search button triggers search and renders results', async () => {
  // Mock the axios.get function to return a sample response
  const mockResponse = {
    data: [
      { trackId: 1, trackName: 'Track 1' },
      { trackId: 2, trackName: 'Track 2' },
    ],
  };
  axios.get.mockResolvedValue(mockResponse);

  // Render the component
  render(<App />);

  // Enter search term and select media type
  const searchInput = screen.getByPlaceholderText('Enter search term');
  fireEvent.change(searchInput, { target: { value: 'Test' } });

  const mediaTypeSelect = screen.getByRole('combobox');
  fireEvent.change(mediaTypeSelect, { target: { value: 'music' } });

  // Click the search button
  const searchButton = screen.getByText('Search');
  fireEvent.click(searchButton);

  // Wait for the search results to be rendered
  await waitFor(() => {
    const result1 = screen.getByText('Track 1');
    const result2 = screen.getByText('Track 2');
    expect(result1).toBeInTheDocument();
    expect(result2).toBeInTheDocument();
  });
});
