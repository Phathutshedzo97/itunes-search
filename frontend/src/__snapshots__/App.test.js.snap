// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders iTunes Search App title 1`] = `
<DocumentFragment>
  <div
    class="container mt-4"
  >
    <h1
      class="mb-4"
    >
      iTunes Search App
    </h1>
    <div
      class="row mb-3"
    >
      <div
        class="col-md-6 col-lg-4"
      >
        <input
          class="form-control"
          placeholder="Enter search term"
          type="text"
          value=""
        />
      </div>
      <div
        class="col-md-4 col-lg-3"
      >
        <select
          class="form-control"
        >
          <option
            value="all"
          >
            All
          </option>
          <option
            value="movie"
          >
            Movie
          </option>
          <option
            value="music"
          >
            Music
          </option>
        </select>
      </div>
      <div
        class="col-md-2 col-lg-2"
      >
        <button
          class="btn btn-primary w-100"
        >
          Search
        </button>
      </div>
    </div>
    <div />
    <div
      class="mt-4"
    >
      <h2>
        Favorites
      </h2>
    </div>
  </div>
</DocumentFragment>
`;
